core/observables/observable.py

```diff
@@ -128,10 +128,18 @@ def add_text(cls, text, tags=[]):
             A saved Observable instance.
 
         """
-        o = Observable.guess_type(text).get_or_create(value=text)
-        if tags:
-            o.tag(tags)
-        return o
+        if force_type:
+            observable_type = get_subclass(Observable, force_type)
+        else:
+            observable_type = Observable.guess_type(text)
+
+        if observable_type:
```

Comment:
I think we should hard-fail here. If `observable_type` is `None` it means that something has gone wrong somewhere (either nothing could be guessed from the text, or the "forced" subclass does not exist). We should raise an exception to highlight the fact that this is not behaving as expected.

I think raising a `GenericYetiError` (see [this](https://github.com/yeti-platform/yeti/blob/master/core/errors.py)) would be appropriate here.